### PR TITLE
Fix alignment of mmap results

### PR DESCRIPTION
**Pull request by sbc100**
Use mmapAlloc for anonymous as well as file-backed
mmap allocations.  Always return pages-aligned regions.

### CODE REVIEW

File: emcc.py [hunk 1]
```diff
@@ -1815,6 +1815,7 @@ def default_setting(name, new_default):
     # helping people see when they should have enabled EXIT_RUNTIME)
     if settings.EXIT_RUNTIME or settings.ASSERTIONS:
       settings.EXPORTED_FUNCTIONS += ['_fflush']
+    settings.EXPORTED_FUNCTIONS += ['_memalign']
```
Comment (kripken) [inline emcc.py:1818]: unintended? this sounds quite heavyweight otherwise, it will end up adding to a lot of programs.

Comment (sbc100) [reply emcc.py:1818]: I can instead make mmapAlloc fail at runtime if the export is missing?  (like makeMalloc does?) ,  but this is more technically correct, the FS library does transitively depend on this.   

I guess I was assuming programs that include FILESYSTEM are already quite bloated/large (at least their JS side will be, no?)

Comment (kripken) [reply emcc.py:1818]: Shouldn't the deps_info change suffice for this? It's not that all FILESYSTEM programs depend on mmapAlloc, just ones that do mmap.

FILESYSTEM using code does get some bloat, that is true, but I don't think we need to increase that here unless I am missing something.

Comment (sbc100) [reply emcc.py:1818]: In theory yes, in practice the JS code for the file system is monolithic, including FS, also includes all the mmap implementations.   These may well be unused at runtime, but closure will see the call to memalign and complain.      I can use the same trick we do in `makeMalloc` though I think..

Comment (sbc100) [reply emcc.py]: Fixed.

**Review by kripken — APPROVED** [review]

File: src/library.js [hunk 2]
```diff
@@ -3682,6 +3682,23 @@ LibraryManager.library = {
     return x.indexOf('dynCall_') == 0 || unmangledSymbols.includes(x) ? x : '_' + x;
   },
 
+  // Allocate memory for an mmap operation. This allocates space of the right
+  // page-aligned size, and clears the padding.
```
Comment (dschuff) [inline src/library.js:3686]:
```suggestion
  // page-aligned size, and clears the allocated space.
```

**Review by dschuff — APPROVED** [review]

**Pull request merged**